cloudinit/config/cc_growpart.py

```diff
@@ -93,6 +94,53 @@ class RESIZE(object):
 LOG = logging.getLogger(__name__)
 
 
+@lru_cache()
+def is_lvm_lv(devpath):
+    if util.is_Linux():
+        # all lvm lvs will have a realpath as a 'dm-*' name.
+        rpath = os.path.realpath(devpath)
+        if not os.path.basename(rpath).startswith("dm-"):
+            return False
+        out, _ = subp.subp("udevadm", "info", devpath)
+        # lvs should have DM_LV_NAME=<lvmuuid> and also DM_VG_NAME
+        return 'DM_LV_NAME=' in out
+    else:
+        return False
+
+
+@lru_cache()
+def get_pvs_for_lv(devpath):
+    myenv = {'LANG': 'C'}
+
+    if util.is_Linux():
+        try:
+            (out, _err) = subp.subp(["lvm", "lvs", devpath, "--options=vgname",
+                                     "--noheadings"], update_env=myenv)
+            vgname = out.strip()
+        except subp.ProcessExecutionError as e:
+            if e.exit_code != 0:
+                util.logexc(LOG, "Failed: can't get Volume Group information "
+                            "from %s", devpath)
+                raise ResizeFailedException(e) from e
+
+        try:
+            (out, _err) = subp.subp(["lvm", "vgs", vgname, "--options=pvname",
+                                     "--noheadings"], update_env=myenv)
+            pvs = [p.strip() for p in out.splitlines()]
+            if len(pvs) > 1:
+                util.logexc(LOG, "Do not know how to resize multiple Physical"
```

Comment:
This should be probably be a logging.warning since we aren't really adding logexc value here. We also should raise a  ResizeFailedException here right? otherwise I think we fall through to the end of the function and return None which then is used by 
bname = os.path.basename(rpath) which will die with a TypeError here 
```TypeError: expected str, bytes or os.PathLike object, not NoneType```

Reply:
@otubo , @blackboxsw 

> This should be probably be a logging.warning since we aren't really adding logexc value here

I think this was the one sticking point that both Chad and I had.

I think this is at most 'info'.  WARNING means something went wrong.  I'd like to not introduce WARN into someone's logs that was happily using cloud-init and LVM previously.  ie, if someone has been using cloud-init with multiple PVs and they upgrade their cloud-init, they will now start seeing this WARN in their logs, even though nothing change for them (before: it didn't resize this.  after: it didn't resize this).

Thats my $0.02.